src/components/network-chart/index.js

```diff
@@ -48,9 +52,20 @@ type OwnProps = {| ...SizeProps |};
 type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
 
 class NetworkChart extends React.PureComponent<Props> {
-  // This isn't used at the moment, but we need a fixed instance so that a
-  // rerender isn't triggered in VirtualList.
-  _specialItems = [];
+  _memoizedGetSpecialItems = memoize(
+    rightClickedMarkerIndex => {
+      if (rightClickedMarkerIndex !== null) {
+        return [rightClickedMarkerIndex];
+      }
+      return [];
+    },
+    { limit: 1 }
+  );
+
+  _getSpecialItems = () => {
+    const { rightClickedMarkerIndex } = this.props;
+    return this._memoizedGetSpecialItems(rightClickedMarkerIndex);
+  };
```

Comment:
I departed from how we do it in other panels. I think we should do it this way in other panels too, so that we don't need to compute it in `componentWillReceiveProps` and friends.